IronPythonExtension/IronPythonExtension.cs

```diff
@@ -91,4 +75,35 @@ public void Shutdown()
             // Do nothing for now
         }
     }
+    internal class IsolatedPythoContext : AssemblyLoadContext
+    {
+        private AssemblyDependencyResolver resolver;
+
+        public IsolatedPythoContext(string libPath)
+        {
+            resolver = new AssemblyDependencyResolver(libPath);
+        }
+
+        protected override Assembly Load(AssemblyName assemblyName)
+        {
+            string assemblyPath = resolver.ResolveAssemblyToPath(assemblyName);
+            if (assemblyPath != null)
+            {
+                return LoadFromAssemblyPath(assemblyPath);
+            }
+
+            return null;
+        }
+
+        protected override IntPtr LoadUnmanagedDll(string unmanagedDllName)
+        {
+            string libraryPath = resolver.ResolveUnmanagedDllToPath(unmanagedDllName);
+            if (libraryPath != null)
+            {
+                return LoadUnmanagedDllFromPath(libraryPath);
+            }
+
+            return IntPtr.Zero;
+        }
```

Comment:
Are these 2 methods required to be overridden for the call to `alc.LoadFromAssemblyName` to work as expected?

Reply:
These overridden methods don't look like they're doing anything special so I wonder if we can get away by just initializing a new `AssemblyLoadContext` base object and calling `LoadFromAssemblyName` on it. 🤔

Reply:
let me take a look, I think the base class implementation does even less...

Reply:
we have the implement them, this implementation is basically from the sample plugin architecture here:
https://learn.microsoft.com/en-us/dotnet/core/tutorials/creating-app-with-plugin-support

the base class doesn't do anything - it just returns
`return IntPtr.Zero;`